src/components/Image.astro

```diff
@@ -42,29 +50,47 @@ const {
 {
   src && (
     <>
-      {!width && !height && (
-        <Root
-          class:list="image"
-          if={src}
-          {src}
-          {alt}
-          as="img"
-          {frame}
-          {position}
-          {radius}
-          {...rest}
-        />
-      )}
-      {width && height && (
+      {isLocalAsset ? (
         <AstroImage
```

Comment:
You use the AstroImage twice, here and below. You can just include in in the check, something like
`isLocalAsset || (!isLocalAsset && width && height)`